src/main/java/seedu/address/model/entry/util/EntryAutofill.java

```diff
@@ -104,23 +101,25 @@ public EntryAutofill extractFromHtml(String html) {
                     )); // desc 1st choice - extract description
 
         }
-        return this;
     }
 
-    /**
-     * Gets the best title.
-     * @return autofilled title if original title is empty, else original title
-     */
-    public Title getTitle() {
-        return title.isEmpty() ? titleCandidate.get() : title;
+    /** Extract candidate by parsing RSS/Atom feed metadata. */
+    public void extractFromFeed(SyndFeed feed) {
```

Comment:
Shall we also add a `extractFromFeedUrl` method to be called before `extractFromFeed`?
For example, `http://rss.nytimes.com/services/xml/rss/nyt/HomePage.xml` will try a `Title` of `rss.nytimes.com` and `Description` of `HomePage`.
```java
public void extractFromFeedUrl(URL url) {
        if (noTitleOrNoDescription()) {
            String baseName = Files.getNameWithoutExtension(url.getPath())
                    .replaceAll("\n", "") // remove newline chars
                    .replaceAll("\r", "") // remove carriage return chars
                    .replaceAll("[^a-zA-Z0-9]+", " ") // replace special chars with spaces
                    .trim();
            titleCandidate.tryout(url.getHost()); // description - host name
            descriptionCandidate.tryout(WordUtils.capitalizeFully(baseName)); // title - cleaned up base name
        }
    }
```

Reply:
The code looks exactly like `extractFromUrl`, maybe I just use that?

Reply:
It's the opposite for title and description if you look carefully :D